app/core/projections.py

```
@@ -0,0 +1,90 @@
+    """Registered projection handler: name, version, subject_type, handler."""
+
+    name: str
+    version: int
+    subject_type: str | None
+    handler: Callable[..., Awaitable[dict[str, Any]]]
+
+
+class ProjectionRegistry:
+    """Singleton registry of all @projection-decorated handlers."""
+
+    _handlers: dict[tuple[str, int], ProjectionRegistration] = {}
```

Ruff (0.15.5)

[warning] Mutable default value for class attribute

(RUF012)